heimlig/src/common/jobs.rs

```diff
@@ -14,41 +15,138 @@ pub enum Error {
     KeyStore(keystore::Error),
 }
 
+/// Shared memory representation.
+#[derive(Eq, PartialEq, Debug)]
+pub struct ExternalMemory {
+    /// Pointer to memory.
+    ptr: *const u8,
+    /// Memory size.
+    size: usize,
+}
+
+impl ExternalMemory {
+    pub fn new(ptr: *const u8, size: usize) -> Self {
+        Self { ptr, size }
+    }
+
+    pub fn from_slice(data: &[u8]) -> Self {
+        Self::new(data.as_ptr(), data.len())
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
```

Comment:
If we find ourselves getting tired of calling `as_slice` explicitly we can think about implementing the `Deref` trait for `ExternalMemory`.